app/models/invoice.rb

```diff
@@ -3,6 +3,19 @@ class Invoice < ApplicationRecord
   belongs_to :merchant
 
   has_many :transactions
+  has_many :invoice_items
 
   validates :status, presence: true
+
+  def total
+    invoice_items.reduce(0) do |total, item|
+      total += (item.unit_price * item.quantity)
+    end
+  end
+
+  def successful?
```

Comment:
These methods are here to support the query. Should I be doing this as part of the query @s-espinosa?